source/PlayerInfo.cpp

```diff
@@ -2354,7 +2356,35 @@ void PlayerInfo::StepMissions(UI *ui)
 		else if(mission.CanComplete(*this))
 			RemoveMission(Mission::COMPLETE, mission, ui);
 		else if(mission.Destination() == GetPlanet() && !freshlyLoaded)
+		{
 			mission.Do(Mission::VISIT, *this, ui);
+			
+			// On visit dialogs are handled separately as to avoid a player
+			// getting spammed by on visit dialogs if they are stacking jobs
+			// from the same destination.
+			map<Mission::Trigger, MissionAction> actions = mission.GetActions();
+			auto ait = actions.find(Mission::VISIT);
+			if(ait != actions.end())
+			{
+				if(visitText.empty())
+				{
+					map<string, string> subs;
+					subs["<first>"] = firstName;
+					subs["<last>"] = lastName;
+					if(Flagship())
+						subs["<ship>"] = Flagship()->Name();
+					visitText = Format::Replace(ait->second.DialogText(), subs);
+				}
+				else
+					++visitDialogs;
+			}
+		}
+	}
+	if(!visitText.empty())
+	{
+		if(visitDialogs)
+			visitText += "\n\tYou have " + Format::Number(visitDialogs) + " other incomplete " + ((visitDialogs > 1) ? "missions" : "mission") + " at this location.";
```

Comment:
I'm not sure about the word `incomplete`. Maybe a word like `unfinished` would fit better?

Similar for `mission/missions`, maybe use the word `job/jobs` there?

Reply:
Not too keen on distinguishing jobs from other missions. Jobs and spaceport missions both go under the active missions tab, so it isn't out of the ordinary to refer to jobs as missions. Plus, I get the sense that using the term `job(s)`, there'd also need to be a case for if you had a spaceport mission *and* a job on visit.